src/blk/kernel/KernelDevice.cc

```diff
@@ -133,8 +133,24 @@ int KernelDevice::open(const string& p)
   int r = 0, i = 0;
   dout(1) << __func__ << " path " << path << dendl;
 
+  struct stat statbuf;
+  bool is_block;
+  r = stat(path.c_str(), &statbuf);
+  if (r != 0) {
+    derr << __func__ << " stat got: " << cpp_strerror(r) << dendl;
+    goto out_fail;
+  }
+  is_block = (statbuf.st_mode & S_IFMT) == S_IFBLK;
   for (i = 0; i < WRITE_LIFE_MAX; i++) {
-    int fd = ::open(path.c_str(), O_RDWR | O_DIRECT);
+    int flags = 0;
+    if (lock_exclusive && is_block && (i == 0)) {
+      // If opening block device use O_EXCL flag.
+      // For block devices ::flock is not enough,
+      // since 2 different inodes with same major/minor can be locked.
```

Comment:
Should we still bother with OFD locks for block devices then?  IOW would it make sense to change the condition for calling `_lock` below from `lock_exclusive` to `lock_exclusive && !is_block`?